common/flatpak-dir.c

```diff
@@ -2649,8 +2649,14 @@ flatpak_dir_get_config (FlatpakDir *self,
                         const char *key,
                         GError    **error)
 {
-  GKeyFile *config = ostree_repo_get_config (self->repo);
-  g_autofree char *ostree_key = g_strconcat ("xa.", key, NULL);
+  GKeyFile *config;
+  g_autofree char *ostree_key = NULL;
+
+  if (!flatpak_dir_ensure_repo (self, NULL, error))
```

Comment:
This should call flatpak_dir_maybe_ensure_repo and then properly handle the self->repo == NULL case (by behaving as if the key is unset). Because this happens if there is /var/lib/flatpak (initially) then ensure_repo() will fail with permissions, when we could really still support get_config().